include/bls12-381/fp.hpp

```diff
@@ -36,6 +36,19 @@ class fp
     int64_t cmp(const fp& e) const;
     bool equal(const fp& e) const;
     bool sign() const;
+    
+    fp add(const fp& e) const;
+    void addAssign(const fp& e);
+    fp dbl() const;
+    void doubleAssign();
+    fp subtract(const fp& e) const;
+    void subtractAssign(const fp& e);
+    fp negate() const;
+    fp multiply(const fp& e) const;
+    void multiplyAssign(const fp& e);
```

Comment:
What about using the more intuitive versions:

```
    fp operator+(const fp& e) const;
    void operator+=(const fp& e);
    fp dbl() const;
    void doubleAssign();
    fp operator-(const fp& e) const;
    void operator-=(const fp& e);
    fp operator-() const;
    fp operator*(const fp& e) const;
```

Reply:
These functions are here to make them align with fp2, fp6 etc.

The will not really be used in our code for performance reason.

If we want operators for fp, then maybe we also want operators for fp2 fp6 and maybe fp12. Then the problem becomes is this really necessary...

Reply:
Personally, I prefer using the intuitive operators, rather than invent new function names whose usage is less straightforward imo.
But I am not requesting this to be changed.